models/DataObject/Concrete.php

```diff
@@ -736,11 +724,9 @@ public function __wakeup(): void
         parent::__wakeup();
 
         // renew object reference to other object aware fields
-        foreach ($this->__objectAwareFields as $objectAwareField => $exists) {
-            if (isset($this->$objectAwareField)) {
-                if ($this->$objectAwareField instanceof ObjectAwareFieldInterface) {
-                    $this->$objectAwareField->setObject($this);
-                }
+        foreach (get_object_vars($this) as $propertyValue) {
+            if ($propertyValue instanceof ObjectAwareFieldInterface) {
+                $propertyValue->setObject($this);
             }
```

Comment:
```suggestion
        if(!$this->isInDumpState()) {
            // renew object reference to other object aware fields
            foreach ($this->__objectAwareFields as $objectAwareField => $exists) {
                if (isset($this->$objectAwareField)) {
                    if ($this->$objectAwareField instanceof ObjectAwareFieldInterface) {
                        $this->$objectAwareField->setObject($this);
                    }
                }
            }
        } else {
            // renew object reference to other object aware fields
            // similar as __wakeup() but now test ALL fields
            // this is necessary for restoring old versions from Pimcore <= 10, see also
            // https://github.com/pimcore/pimcore/pull/16419
            foreach (get_object_vars($this) as $propertyValue) {
                if ($propertyValue instanceof ObjectAwareFieldInterface) {
                    $propertyValue->setObject($this);
                }
            }
        }
```

Reply:
@fashxp @youwe-petervanderwal my suggestion should fix the issue while not impacting performance. WDYT?

Reply:
@brusch Nice, thanks for your comment! I didn't knew about the `isInDumpState()` method. This should (theoratically) produce the desired result. I'll try and debug it locally to be sure it has the desired behavior and then I'll update my PR (later today or tomorrow)

cc @fashxp

Reply:
@fashxp @brusch I've applied the suggested code change (needed one fix because `parent::__wakeup()` updates the `inDumpState` value to `false) and I've verified with debugger that it follows the optimized flow when reloading objects from (redis) cache. 

This solution works like a charm :)